File: Service/ScanCare/RMA.Service.ScanCare.Contracts/Entities/Document.cs
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

using System;
using System.Collections.Generic;

namespace RMA.Service.ScanCare.Contracts.Entities
{
    public class Document
    {
        public int Id { get; set; }
        public int DocTypeId { get; set; }
        public string SystemName { get; set; }
        public string DocumentUri { get; set; }
        public string VerifiedBy { get; set; }
        public DateTime? VerifiedByDate { get; set; }
        public string FileHash { get; set; }
        public string FileName { get; set; }
        public string FileExtension { get; set; }
        public DocumentStatusEnum? DocumentStatus { get; set; }
        public string FileAsBase64 { get; set; }
        public Dictionary<string, string> Keys { get; set; }
        public string DocumentTypeName { get; set; }
        public DocumentSetEnum? DocumentSet { get; set; }
        public string CreatedBy { get; set; }
        public DateTime? CreatedDate { get; set; }
        public string MimeType { get; set; }
        public bool DocumentExist { get; set; }
        public bool Required { get; set; }
        public string DocumentDescription { get; set; }
        public bool IsMemberVisible { get; set; }
    }
}

File: Web/RMA-Member-Portal/memberportal-app/src/app/shared/components/document-management/document.ts
import { DocumentStatusEnum } from "../../enums/document-status.enum";

export class Document {
  id: number;
  docTypeId: number;
  systemName: string;
  documentUri?: string;
  verifiedByDate?: Date;
  documentTypeName: string;
  verifiedBy?: string;
  fileHash?: string;
  fileName: string;
  fileExtension: string;
  documentStatus: DocumentStatusEnum;
  documentStatusText: string;
  documentSet: string;
  mimeType: string;
  documentExist: boolean;
  required: boolean;
  documentDescription: string;
  isMemberVisible: boolean;

  fileAsBase64: string;
  keys: { [key: string]: string };
  createdBy: string;
  createdDate: Date;
}
